numpy/core/tests/test_numeric.py

```diff
@@ -48,6 +48,14 @@ def test_reshape_from_zero(self):
         assert_array_equal(Ar, np.zeros((2, 1), Ar.dtype))
         assert_equal(A.dtype, Ar.dtype)
 
+    def test_negativeresize(self):
+        A = np.arange(0, 10, dtype=np.float32)
+        new_shape = (10,-1)
+        err_msg = (r"new_shape can't have negative values."
+                   r" found: new_shape=(10,-1)")
+        with pytest.raises(ValueError, match=err_msg):
```

Comment:
`match="negative"` would probably be enough here - matching the full message makes the test overly brittle to wording changes.

Reply:
Done!

Reply:
I dont know why the tests are failing with the current fix! Can you please guide me through it? Thanks for your replies!

Reply:
You can see the failing test result here:

https://dev.azure.com/numpy/numpy/_build/results?buildId=6722&view=ms.vss-test-web.build-test-results-tab&runId=1078974&resultId=100000&paneView=debug

Reply:
There's a bug in the error handling, opened gh-15039

Reply:
Ohk! Thanks!